### PR TITLE
Patch system-setting to not reference CLLocationManager

### PR DESCRIPTION
**Pull request by gavrix**
system-setting library checks many different APIs for if it's enabled or not, including `CLLocationManager`.
Although there is never an attempt to ENABLE location services, apple may statically check _any_ references to CLLocationManager symbol and flag this activity. 

Solved to excluding this API reference (and statically returning NO instead of `[CLLocationManager isEnabled]`. Since this requires modification of a library code, patched it using `patch-package`.

### CODE REVIEW

File: patches/react-native-system-setting+1.7.4.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
+diff --git a/node_modules/react-native-system-setting/ios/RTCSystemSetting.m b/node_modules/react-native-system-setting/ios/RTCSystemSetting.m
+index c94aaad..e2f44a7 100644
+--- a/node_modules/react-native-system-setting/ios/RTCSystemSetting.m
++++ b/node_modules/react-native-system-setting/ios/RTCSystemSetting.m
+@@ -8,7 +8,7 @@
+ 
+ #import "RTCSystemSetting.h"
+ #import <SystemConfiguration/CaptiveNetwork.h>
+-#import <CoreLocation/CoreLocation.h>
++//#import <CoreLocation/CoreLocation.h>
```
Comment (honkfestival) [inline patches/react-native-system-setting+1.7.4.patch:10]: We can just delete this right?

File: patches/react-native-system-setting+1.7.4.patch [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+@@ -122,7 +122,8 @@ +(BOOL)requiresMainQueueSetup{
+ }
+ 
+ RCT_EXPORT_METHOD(isLocationEnabled:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
+-    resolve([NSNumber numberWithBool:[CLLocationManager locationServicesEnabled]]);
++    resolve(@NO);
++//    resolve([NSNumber numberWithBool:[CLLocationManager locationServicesEnabled]]);
+ }
+ 
+ RCT_EXPORT_METHOD(switchBluetooth){
```
Comment (gavrix) [inline patches/react-native-system-setting+1.7.4.patch:23]: This is library code change in question.

File: patches/react-native-system-setting+1.7.4.patch [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ #import <CoreTelephony/CTTelephonyNetworkInfo.h>
+ #import <ifaddrs.h>
+ #import <net/if.h>
+@@ -122,7 +122,8 @@ +(BOOL)requiresMainQueueSetup{
+ }
+ 
+ RCT_EXPORT_METHOD(isLocationEnabled:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
+-    resolve([NSNumber numberWithBool:[CLLocationManager locationServicesEnabled]]);
++    resolve(@NO);
++//    resolve([NSNumber numberWithBool:[CLLocationManager locationServicesEnabled]]);
```
Comment (honkfestival) [inline patches/react-native-system-setting+1.7.4.patch:20]: Same here

File: patches/react-native-system-setting+1.7.4.patch (hunk 1 above)
Comment (gavrix) [reply patches/react-native-system-setting+1.7.4.patch:10]: since it's a patch it doesn't really matter. I can delete comments.

**Pull request merged**